src/lib.rs

```diff
@@ -137,14 +137,15 @@ mod tokio_tests {
 
         tokio::spawn(async move {
             let (stream, _) = echo.accept().await.unwrap();
-            let mut stream = AsyncBincodeStream::<_, usize, usize, _>::from(stream).for_async();
-            let (r, w) = stream.tcp_split();
+            let stream =
+                AsyncBincodeStream::<_, usize, usize, _>::from(BufStream::new(stream)).for_async();
+            let (w, r) = stream.split();
```

Comment:
Oof, the need to use the (much less efficient) `.split` here instead of `.tcp_split` is unfortunate. I think what this really implies is that we should have `AsyncBincodeStream` support keeping the reader and writer separately so that we can split _before_ constructing the `AsyncBincodeStream`. That's probably a change for down the line instead though.